BankAccounts/src/com/company/AccountService.java

```diff
@@ -0,0 +1,47 @@
+package com.company;
+
+import java.io.FileWriter;
+
+public class AccountService {
+
+    public void transfer(Account accountA, Account accountB, double amount) {
+        synchronized(accountA)
+        {
+            withdraw(accountA, amount);
+            System.out.print(amount+" is withdrawn from account a\n");
+
+            try{Thread.sleep(500);}
+            catch(Exception e){System.out.println(e);}
+
+            synchronized(accountB)
+            {
+                deposit(accountB, amount);
+                System.out.print(amount+" is deposited into account b\n");
+            }
+        }
+    }
+
+    private double deposit(Account account, double amount) {
+        try{
+            account.setBalance(account.getBalance() + amount);
+            FileWriter myWriter = new FileWriter(account.getAccountName() + ".txt");
+            myWriter.write(String.valueOf(account.getBalance()));
+            Thread.sleep(100);
+        } catch (Exception e) {
+            System.out.println("Something went wrong");
+        }
+        return account.getBalance();
+    }
+
+    private double withdraw(Account account, double amount) {
+        try{
+            account.setBalance(account.getBalance() - amount);
+            FileWriter myWriter = new FileWriter(account.getAccountName() + ".txt");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *myWriter*. Execution paths that do not contain closure statements were detected. Ensure *myWriter* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).